File: Leetcode/921. Minimum Add to Make Parentheses Valid/solution.py
class Solution:
    def minAddToMakeValid(self, S: str) -> int:
        stack = []
        for char in S:
            if char == '(':
                stack.append(char)
            elif stack and stack[-1] == '(':
                stack.pop()
            else:
                stack.append(char)

        return len(stack)

File: Leetcode/921. Minimum Add to Make Parentheses Valid/solution.rs
impl Solution {
    pub fn min_add_to_make_valid(s: String) -> i32 {
        let mut stack = Vec::new();

        for c in s.chars() {
            if stack.len() > 0 && stack[stack.len() - 1] == '(' && c == ')' {
                stack.pop();
            } else {
                stack.push(c);
            }
        }
        stack.len() as i32
    }
}
